### DISCUSSION TITLE
Possible Issue With Crazyswarm, ROS 1 and cmdVel()

### DISCUSSION THREAD

**Discussion by BBooda** [bitcraze/discussions#594] [Q&A]
Hello everyone, 

I have updated a crazyflie 2 with the latest firmware and I am trying to use crazyswarm and the ROS 1 cmdVel() command to send velocity commands to the drone. 

While the radio link works, and the ros service call also works, i.e., it executes the thrust commands to the motors (tested with the helloWorld python script). When I am trying to send cmdVel() commands directly, the drone doesn't respond. I have seen similar discussions relating to this topic and tried the proposed solutions. 

The first solution that I have seen is to first send a couple, or some cmdVel commands with the thrust set as zero. This was to address a possible thrust lock.. And then send other cmdVel commands with thrust ~= 0. But it didn't work. 

When I publish cmdVel() commands, I can also perform a simple ros echo command and see that the commands are indeed published in the ROS 1 layer. 

The other solution that I saw was related to not have the latest firmware, but to me that is not the case. 

Is there something I am missing ? What could be the problem?

For odometry I am using a Vicon system, but I also tried to disable the Vicon/optitrack etc. since I guess it doesn't really need to have the feedback to execute the velocity command. 

Thanks a lot in advance, sorry if this is a duplicate, but I couldn't locate the problem to the existing ones.

**Comment by knmcguire** [comment]
Hi! 

So this would better to post on the Crazyswarm1 discussions as we haven't worked with Crazyswarm1 often enough to find out what is going on: https://github.com/USC-ACTLab/crazyswarm/discussions

But if you are not married with ROS1, I would highly advise you to move to ROS 2 with Crazyswarm2: https://github.com/IMRCLab/crazyswarm2/discussions. Crazyswarm1 will have less and less support in the future and Crazyswarm2 is more regularly tested now, eventhough not all functionalities have been ported
